src/rpy_diffusive_flux.cc

```diff
@@ -33,9 +39,23 @@ void RPYDiffusiveFlux::compute(std::shared_ptr<GrandPotential> grand, std::share
         {
         throw std::invalid_argument("Spherical geometry required");
         }
-
+    
+    if(state->getNumFields() > 1)
+        {
+        throw std::invalid_argument("Only one type of particle is supported");
+        }
+        
+    const GridInterpolator& g = *mobility_;
+    
+    double max_x = std::get<0>(g.getBounds());
+    double max_density = std::get<2>(g.getBounds());
+    
     for (const auto &i : state->getTypes())
         {
+        if(diameters_(i) > 1)
```

Comment:
This should be like:
```suggestion
        if(diameters_(i) != 1)
```
since anything other than 1 is not allowed (including < 1). If it triggers an error because the value is close to (but not exactly) 1, we can write this check in a slightly different way.